src/main/java/com/ensao/gi5/lint/rules/NumberOfParametersRule.java

```diff
@@ -0,0 +1,67 @@
+package com.ensao.gi5.lint.rules;
+
+import com.ensao.gi5.lint.constantes.Constantes;
+import com.ensao.gi5.lint.rules.violations.Violation;
+import com.ensao.gi5.lint.rules.violations.ViolationFactory;
+import com.ensao.gi5.lint.util.Utils;
+import com.ensao.gi5.lint.visitor.NumberOfMethodsVisitor;
+import com.ensao.gi5.lint.visitor.NumberOfParametersVisitor;
+import com.ensao.gi5.lint.wrapper.CompilationUnitWrapper;
+import com.github.javaparser.ast.body.ConstructorDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+
+/**
+ * The NumberOfParametersRule class is a Java class that extends the Rule abstract class.
+ * It is used to check the number of parameters within a method or constructor of a compilation unit.
+ *
+ * The "apply" method contains the main logic of the class. It creates two sets, one for "MethodDeclaration"
+ * and one for "ConstructorDeclaration". It then uses two visitors, "NumberOfMethodsVisitor" and
+ * "NumberOfParametersVisitor", to populate these sets with all the methods and constructors respectively,
+ * in the provided "CompilationUnitWrapper" object.
+ *
+ * This rule check the number of parameters in all constructors and methods,
+ * and if it's more than 2 it will raise a violation.
+ * **/
+public class NumberOfParametersRule extends Rule{
```

Comment:
Separation of concerns : the rule applied on both methods and constructors, it has to be split into separate classes.